service/storage_proxy.cc

```diff
@@ -4593,6 +4595,36 @@ future<bool> storage_proxy::cas(schema_ptr schema, shared_ptr<cas_request> reque
     co_return condition_met;
 }
 
+bool storage_proxy::is_likely_to_respond_in_time(const gms::inet_address& ep, clock_type::time_point timeout) {
+    if (fbu::is_me(ep)) {
+        return true;
+    }
+    if (auto last_seen_ep = last_seen(ep)) {
+        auto now = clock_type::now();
+        auto time_without_response = last_seen_ep->last_sent - last_seen_ep->last_responded;
```

Comment:
I'm not sure this is a good metric. Let's say a replica is receiving 1M reqs/sec and responding 1M reqs/sec, but with a latency of 500ms. Is it overloaded or not?

Reply:
I'm also not sure, but the rough idea is that users would be able to impact this metric with custom timeouts. Let's say that we have 1M reqs/sec with a latency of 500ms and a default 5s timeout. According to this code, the node is not qualified as overloaded. However, if the user sets an explicit timeout of 200ms for this request, because otherwise he loses a bid, then the node is qualified as overloaded.